src/stp/Tactic.cpp

```diff
@@ -4,58 +4,67 @@
 
 #include "stp/Tactic.h"
 
+#include <roboteam_utils/Print.h>
+
 namespace rtt::ai::stp {
 
-void Tactic::initialize() noexcept { onInitialize(); }
+void Tactic::initialize() noexcept {
+  for (auto &x : skills) {
+    x->initialize();
+  }
+}
 
 Status Tactic::update(StpInfo const &info) noexcept {
-    if (!info.getBall() || !info.getRobot() || !info.getField()) {
-        RTT_WARNING("Required information missing in the tactic info")
-        currentStatus = Status::Failure;
-        return Status::Failure;
-    }
-
-    // Update skill info
-    auto skill_info = calculateInfoForSkill(info);
-
-    if (skill_info) {
-        // Update the current skill with the new SkillInfo
-        auto status = skills.update(skill_info.value());
-
-        // Call onUpdate on a skill for specific behaviour
-        onUpdate(status);
-    } else {
-        RTT_ERROR("Not all data was present, bad update!")
-    }
-
-    // the tactic will not be reset if it's the first skill
-    if (skills.current_num() != 0 && shouldTacticReset(skill_info.value())) {
-        reset();
-    }
-
-    // Check if the skills are all finished
-    if (skills.finished() || forceTacticSuccess(skill_info.value())) {
-        if (!isEndTactic()) {
-            currentStatus = Status::Success;
-            return Status::Success;
-        }
-        // Make sure we keep executing the last tactic since it is an end tactic
-        skills.reset();
-        currentStatus = Status::Waiting;
-        return Status::Waiting;
+  if (!info.getBall() || !info.getRobot() || !info.getField()) {
+    RTT_WARNING("Required information missing in the tactic info")
+    currentStatus = Status::Failure;
+    return Status::Failure;
+  }
+
+  // Update skill info
+  auto skill_info = calculateInfoForSkill(info);
+
+  if (skill_info) {
+    // Update the current skill with the new SkillInfo
+    auto status = skills.update(skill_info.value());
+
+    (void)status;
```

Comment:
I don't understand how this change works, does it preserve the functionality? And why did you change it?

Reply:
Well, the return of update is never used, but it is marked [[no-discard]] in the state machine. This cast to void basically uses up the return to suppress the compile warning. I did not want to touch the state machine so I opted for this solution.